src/diagnostics/Timer.js

```diff
@@ -0,0 +1,24 @@
+/* @flow */
+import { PureComponent } from 'react';
+
+import type { ChildrenArray } from '../types';
+import timing from '../utils/timing';
+
+type Props = {
+  children: ChildrenArray<*>,
+  name: string,
+};
+
+export default class Timer extends PureComponent<Props> {
+  componentDidMount() {
+    timing.start(this.props.name);
+  }
+
+  componentWillUnmount() {
+    timing.end(this.props.name);
+  }
+
+  render() {
+    return this.props.children;
+  }
+}
```

Comment:
Cool. That looks correct just looking at the code.
We might want to name this more descriptively. 'RenderTimer' sounds like an improvement. Any other alternatives you can come up with?